zookeeper-server/src/test/java/org/apache/zookeeper/server/util/JvmPauseMonitorTest.java

```diff
@@ -0,0 +1,133 @@
+        final Long sleepTime = 444L;
+        final Long warnTH = 5555L;
+        final Long infoTH = 555L;
+
+        ServerConfig serverConfig = new ServerConfig();
+        QuorumPeerConfig quorumPeerConfig = new QuorumPeerConfig();
+
+        Assert.assertEquals(false, serverConfig.isJvmPauseMonitorToRun());
+        Assert.assertEquals(false, quorumPeerConfig.isJvmPauseMonitorToRun());
+
+        Properties zkProp = new Properties();
+        zkProp.setProperty("dataDir", new File("myDataDir").getAbsolutePath());
+        zkProp.setProperty("jvm.pause.monitor", "true");
+        zkProp.setProperty("jvm.pause.sleep.time.ms", sleepTime.toString());
+        zkProp.setProperty("jvm.pause.warn-threshold.ms", warnTH.toString());
+        zkProp.setProperty("jvm.pause.info-threshold.ms", infoTH.toString());
+        try {
+            quorumPeerConfig.parseProperties(zkProp);
+        } catch (IOException | QuorumPeerConfig.ConfigException e) {
+            Assert.fail("Exception while reading config for JvmPauseMonitor");
+        }
+        serverConfig.readFrom(quorumPeerConfig);
+
+        Assert.assertEquals(true, serverConfig.isJvmPauseMonitorToRun());
+        Assert.assertEquals(true, quorumPeerConfig.isJvmPauseMonitorToRun());
+
+        JvmPauseMonitor pauseMonitor = new JvmPauseMonitor(serverConfig);
+        Assert.assertFalse(pauseMonitor.isStarted());
+        pauseMonitor.serviceStart();
+        Assert.assertTrue(pauseMonitor.isStarted());
+
+        Assert.assertEquals(sleepTime, Long.valueOf(pauseMonitor.sleepTimeMs));
+        Assert.assertEquals(warnTH, Long.valueOf(pauseMonitor.warnThresholdMs));
+        Assert.assertEquals(infoTH, Long.valueOf(pauseMonitor.infoThresholdMs));
+    }
+
+    @Test
+    public void testJvmPauseMonitorExceedInfoThreshold() throws InterruptedException {
+        final Long sleepTime = 100L;
+        final Long infoTH = -1L;
+
+        ServerConfig serverConfig = new ServerConfig();
+        QuorumPeerConfig quorumPeerConfig = new QuorumPeerConfig();
+
+        Properties zkProp = new Properties();
+        zkProp.setProperty("dataDir", new File("myDataDir").getAbsolutePath());
+        zkProp.setProperty("jvm.pause.monitor", "true");
+        zkProp.setProperty("jvm.pause.sleep.time.ms", sleepTime.toString());
+        zkProp.setProperty("jvm.pause.info-threshold.ms", infoTH.toString());
+        try {
+            quorumPeerConfig.parseProperties(zkProp);
+        } catch (IOException | QuorumPeerConfig.ConfigException e) {
+            Assert.fail("Exception while reading config for JvmPauseMonitor");
+        }
+        serverConfig.readFrom(quorumPeerConfig);
+
+        JvmPauseMonitor pauseMonitor = new JvmPauseMonitor(serverConfig);
+        pauseMonitor.serviceStart();
+
+        Assert.assertEquals(sleepTime, Long.valueOf(pauseMonitor.sleepTimeMs));
+        Assert.assertEquals(infoTH, Long.valueOf(pauseMonitor.infoThresholdMs));
+
+        Thread.sleep(1000);
```

Comment:
It is better to have a loop and wait at most X seconds for the condition to be true.
This way we will reduce flakyness